src/Storageless/Http/SessionMiddleware.php

```diff
@@ -43,14 +43,13 @@
 
 use function sprintf;
 
-/** @immutable */
-final class SessionMiddleware implements MiddlewareInterface
+final readonly class SessionMiddleware implements MiddlewareInterface
```

Comment:
Careful: `readonly` != `@immutable`. `@immutable` implies `readonly` + no side-effects.

Reply:
I also don't know of other tools that enforce immutability right now, which is why I keep sticking to psalm (for now) :P

Reply:
Annotation restored with the more specific `@psalm-immutable` name